### PR TITLE
Add source roots correctly while generating SAM projects

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

CodeUri Property is required only if the packageType is Zip. (Source: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#sam-function-codeuri), so this change ensures that the codeUri is not accessed when the packageType is Image, thus resolving the error mentioned in the related Github Issue.

## Related Issue(s)

https://github.com/aws/aws-toolkit-jetbrains/issues/2605
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonSamProjectWizard.kt [hunk 1]
```diff
@@ -51,7 +51,10 @@ abstract class PythonSamProjectTemplate : SamAppTemplateBased() {
         indicator: ProgressIndicator
     ) {
         super.postCreationAction(settings, contentRoot, rootModel, indicator)
-        addSourceRoots(rootModel.project, rootModel, contentRoot)
+        if(settings.packagingType != PackageType.IMAGE) {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonSamProjectWizard.kt:54]: image templates still technically have source code (and therefore source roots). should we read out the docker context and use that instead?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonSamProjectWizard.kt:54]: See `software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateUtils#getCodeLocation`

The current method it is using (`software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon.Companion#getCodeUrisFromTemplate`) should be migrated off of and deleted

Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonSamProjectWizard.kt]: Migrated off getCodeUrisFromTemplate from this function, will add  a separate PR for deleting it completely

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt [hunk 2]
```diff
@@ -68,11 +71,28 @@ abstract class SamProjectTemplate {
...
+        val templatePath = Paths.get(template.parent.path)
+        runInEdt {
+            val functions = SamTemplateUtils.findFunctionsFromTemplate(project, template)
+            val functionLocations = functions.map {
+                val codeLocation = SamTemplateUtils.getCodeLocation(template.toNioPath().toAbsolutePath(), it.logicalName)
+                if(codeLocation.startsWith("./")){
+                    templatePath.resolve(codeLocation.substring(2))
+                } else {
+                    templatePath.resolve(codeLocation)
+                }
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt:83]: I dont think you need to check "./", the path API should always relativize it if its not "/"

I would do `templatePath.parent.resolve(codeLocation)`

Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt]: Made the change

**Comment by abrooksv** [comment]
Detekt failures, but look good once you fix those

**Pull request merged**